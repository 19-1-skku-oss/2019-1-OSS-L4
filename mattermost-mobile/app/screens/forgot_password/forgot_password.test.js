// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import ForgotPassword from './forgot_password.js';

jest.mock('react-intl');

describe('ForgotPassword', () => {
    const actions = {
        sendPasswordResetEmail: jest.fn(),
    };

    const baseProps = {
        actions,
    };

    const formatMessage = jest.fn();

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ForgotPassword {...baseProps}/>,
            {context: {intl: {formatMessage}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('snapshot for error on failure of email regex', () => {
        const wrapper = shallow(
            <ForgotPassword {...baseProps}/>,
            {context: {intl: {formatMessage}}},
        );

        wrapper.setState({email: 'bar'});
        wrapper.instance().submitResetPassword();
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('Should call sendPasswordResetEmail', () => {
        const wrapper = shallow(
            <ForgotPassword {...baseProps}/>,
            {context: {intl: {formatMessage}}},
        );

        wrapper.setState({email: 'test@test.com'});
        wrapper.instance().submitResetPassword();
        expect(actions.sendPasswordResetEmail).toBeCalledWith('test@test.com');
    });

    test('match snapshot after success of sendPasswordResetEmail', async () => {
        const sendPasswordResetEmail = async () => {
            return {
                data: {},
            };
        };
        const wrapper = shallow(
            <ForgotPassword
                {...baseProps}
                actions={{
                    ...actions,
                    sendPasswordResetEmail,
                }}
            />,
            {context: {intl: {formatMessage}}},
        );

        wrapper.setState({email: 'test@test.com'});
        wrapper.instance().submitResetPassword();
        await sendPasswordResetEmail();
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
