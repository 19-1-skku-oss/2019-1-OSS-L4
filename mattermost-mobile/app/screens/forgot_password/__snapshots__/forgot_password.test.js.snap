// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword match snapshot after success of sendPasswordResetEmail 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 15,
          "paddingVertical": 50,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../dist/assets/images/logo.png",
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#dff0d8",
            "borderColor": "#d6e9c6",
            "marginTop": 15,
            "padding": 10,
          }
        }
      >
        <FormattedText
          defaultMessage="If the account exists, a password reset email will be sent to:"
          id="password_send.link"
          style={
            Object {
              "color": "#3c763d",
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#3c763d",
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          test@test.com
        </Text>
        <FormattedText
          defaultMessage="Please check your inbox."
          id="password_send.checkInbox"
          style={
            Array [
              Object {
                "color": "#3c763d",
              },
              Object {
                "paddingTop": 15,
              },
            ]
          }
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`ForgotPassword should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 15,
          "paddingVertical": 50,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../dist/assets/images/logo.png",
          }
        }
      />
      <View>
        <FormattedText
          defaultMessage="To reset your password, enter the email address you used to sign up"
          id="password_send.description"
          style={
            Array [
              Object {
                "color": "#777",
                "fontSize": 16,
                "fontWeight": "300",
                "lineHeight": 22,
                "marginBottom": 15,
                "textAlign": "center",
              },
              Object {
                "paddingTop": 15,
              },
            ]
          }
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          blurOnSubmit={false}
          disableFullscreenUI={true}
          keyboardType="email-address"
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "stretch",
              "borderColor": "gainsboro",
              "borderRadius": 3,
              "borderWidth": 1,
              "fontSize": 16,
              "height": 45,
              "marginBottom": 5,
              "marginTop": 5,
              "paddingLeft": 10,
            }
          }
          underlineColorAndroid="transparent"
        />
        <Button
          containerStyle={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "borderColor": "#2389D7",
              "borderRadius": 3,
              "borderWidth": 1,
              "marginTop": 10,
              "padding": 15,
            }
          }
          disabled={true}
          onPress={[Function]}
        >
          <FormattedText
            defaultMessage="Reset my password"
            id="password_send.reset"
            style={
              Array [
                Object {
                  "color": "#2389D7",
                  "fontSize": 17,
                  "textAlign": "center",
                },
              ]
            }
          />
        </Button>
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;

exports[`ForgotPassword snapshot for error on failure of email regex 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingHorizontal": 15,
          "paddingVertical": 50,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../dist/assets/images/logo.png",
          }
        }
      />
      <View>
        <FormattedText
          defaultMessage="To reset your password, enter the email address you used to sign up"
          id="password_send.description"
          style={
            Array [
              Object {
                "color": "#777",
                "fontSize": 16,
                "fontWeight": "300",
                "lineHeight": 22,
                "marginBottom": 15,
                "textAlign": "center",
              },
              Object {
                "paddingTop": 15,
              },
            ]
          }
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          blurOnSubmit={false}
          disableFullscreenUI={true}
          keyboardType="email-address"
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "stretch",
              "borderColor": "gainsboro",
              "borderRadius": 3,
              "borderWidth": 1,
              "fontSize": 16,
              "height": 45,
              "marginBottom": 5,
              "marginTop": 5,
              "paddingLeft": 10,
            }
          }
          underlineColorAndroid="transparent"
        />
        <Button
          containerStyle={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "borderColor": "#2389D7",
              "borderRadius": 3,
              "borderWidth": 1,
              "marginTop": 10,
              "padding": 15,
            }
          }
          disabled={false}
          onPress={[Function]}
        >
          <FormattedText
            defaultMessage="Reset my password"
            id="password_send.reset"
            style={
              Array [
                Object {
                  "color": "#2389D7",
                  "fontSize": 17,
                  "textAlign": "center",
                },
              ]
            }
          />
        </Button>
      </View>
    </View>
  </TouchableWithoutFeedback>
</View>
`;
